static/scripts/dashboard/dashboard2Ctrl.js

```diff
@@ -1145,12 +1145,17 @@
       //console.log('Modal Back to old values ' , $scope.vm.initializeForm[fromCopy]);
     }
 
-    function showInstallLocationModal(sectionName) {
+    function showInstallLocationModal(sectionName, clickType) {
       var old = sectionName + '_old';
+      //console.log('get values', $scope.vm.initializeForm[sectionName]);
 
       if (sectionName === 'secrets') {
         $scope.vm.initializeForm[old] =  angular.copy($scope.vm.initializeForm.secrets);
-        $('#secrets-location-modal').modal('show');
+        if (clickType === 'dropdown' && $scope.vm.initializeForm.secrets.initType === 'admiral') {
```

Comment:
if $scope.vm.initializeForm.secrets.initType !== 'admiral'
we can get rid of the else then

Reply:
We are opening the modal in the else portion. If it is == admiral, then we do not need the add node popup.

Reply:
Need to open the modal either from the dropdown or from the button. Both are required for easier usability. So open modal only if New/Existing is pressed. Or open if the button is clicked directly.

Reply:
ok